### ISSUE TITLE
aws-batch-alpha: Parameter Store secrets not supported in new version

### ISSUE THREAD

**Issue by jacob-swanson** [aws/aws-cdk#26339]
### Describe the bug

The older version of aws-batch-alpha allowed for secrets to be defined using either Secrets Manager or Parameter Store.

In the old `JobDefinition` construct, `secrets` was defined as:
```
readonly secrets?: {
    [key: string]: ecs.Secret;
};
``` 
Which works with either Secrets Manager or Parameter Store.

But the new `EcsContainerDefinitionProps` defines `secrets` as
```
readonly secrets?: {
    [envVarName: string]: secretsmanager.ISecret;
};
``` 
Which is just Secrets Manager.

### Expected Behavior

I should be able to create a job definition using secrets from either Secrets Manager or Parameter Store.

### Current Behavior

Job definitions cannot use secrets from parameter store.

### Reproduction Steps

Snippet of old, working code:
```typescript
new batch.JobDefinition(this, "JobDefinition", {
    container: {
        secrets: {
            ENVAR_NAME: ecs.Secret.fromSsmParameter(ssm.StringParameter.fromStringParameterName(this, "StringParameter", "stringParameterName"))
        }
    }
})
```

### CDK CLI Version

2.87.0 (build 9fca790)

### Node.js Version

v18.16.1

### OS

Pop!_OS 22.04 LTS

### Language

Typescript

**Comment by peterwoodworth** [comment]
Yep, looks like this is the case. @comcalvi something to take a look at, not sure how you would like to go about this.

**Comment by comcalvi** [comment]
ah, I missed that it could also be parameter store
